### ISSUE TITLE
CircuitBreakerMTSpec: must recover and reset the breaker after the reset timeout

### ISSUE THREAD

**Issue by andreaTP** [akka/akka#30476]
https://github.com/akka/akka/runs/3242990148?check_suite_focus=true#step:5:2424

```
[info] - must recover and reset the breaker after the reset timeout *** FAILED *** (4 seconds, 3 milliseconds)
[info]   java.util.concurrent.TimeoutException: Timeout of 4 seconds with time factor of 1.0
[info]   at akka.testkit.TestLatch.ready(TestLatch.scala:45)
[info]   at akka.testkit.TestLatch.ready(TestLatch.scala:37)
[info]   at scala.concurrent.Await$.$anonfun$ready$1(package.scala:174)
[info]   at scala.concurrent.BlockContext$DefaultBlockContext$.blockOn(BlockContext.scala:62)
[info]   at scala.concurrent.Await$.ready(package.scala:124)
[info]   at akka.pattern.CircuitBreakerMTSpec.f$proxy4$5(CircuitBreakerMTSpec.scala:91)
[info]   at akka.pattern.CircuitBreakerMTSpec.$init$$$anonfun$5$$anonfun$4(CircuitBreakerMTSpec.scala:103)
Coroner Thread Count started at 18, ended at 10, peaked at 24 in akka.pattern.CircuitBreakerMTSpec
[info]   at org.scalatest.Transformer.apply$$anonfun$1(Transformer.scala:22)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:31)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:21)
[info]   at org.scalatest.wordspec.AnyWordSpecLike$$anon$3.apply(AnyWordSpecLike.scala:1105)
[info]   at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
[info]   at org.scalatest.TestSuite.withFixture$(TestSuite.scala:138)
[info]   at akka.testkit.AkkaSpec.withFixture(AkkaSpec.scala:54)
[info]   at org.scalatest.wordspec.AnyWordSpecLike.invokeWithFixture$1(AnyWordSpecLike.scala:1111)
[info]   at org.scalatest.wordspec.AnyWordSpecLike.runTest$$anonfun$1(AnyWordSpecLike.scala:1115)
[info]   at org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
[info]   at org.scalatest.wordspec.AnyWordSpecLike.runTest(AnyWordSpecLike.scala:1115)
[info]   at org.scalatest.wordspec.AnyWordSpecLike.runTest$(AnyWordSpecLike.scala:44)
[info]   at akka.testkit.AkkaSpec.runTest(AkkaSpec.scala:54)
[info]   at org.scalatest.wordspec.AnyWordSpecLike.runTests$$anonfun$1(AnyWordSpecLike.scala:1174)
[info]   at org.scalatest.SuperEngine.traverseSubNodes$2$$anonfun$1(Engine.scala:413)
[info]   at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[info]   at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[info]   at scala.collection.immutable.List.foreach(List.scala:333)
[info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:429)
[info]   at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:390)
[info]   at org.scalatest.SuperEngine.traverseSubNodes$2$$anonfun$1(Engine.scala:427)
[info]   at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[info]   at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[info]   at scala.collection.immutable.List.foreach(List.scala:333)
[info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:429)
[info]   at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:396)
[info]   at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:475)
[info]   at org.scalatest.wordspec.AnyWordSpecLike.runTests(AnyWordSpecLike.scala:1174)
[info]   at org.scalatest.wordspec.AnyWordSpecLike.runTests$(AnyWordSpecLike.scala:44)
[info]   at akka.testkit.AkkaSpec.runTests(AkkaSpec.scala:54)
[info]   at org.scalatest.Suite.run(Suite.scala:1112)
[info]   at org.scalatest.Suite.run$(Suite.scala:563)
[info]   at akka.testkit.AkkaSpec.org$scalatest$wordspec$AnyWordSpecLike$$super$run(AkkaSpec.scala:54)
[info]   at org.scalatest.wordspec.AnyWordSpecLike.run$$anonfun$1(AnyWordSpecLike.scala:1219)
[info]   at org.scalatest.SuperEngine.runImpl(Engine.scala:535)
[info]   at org.scalatest.wordspec.AnyWordSpecLike.run(AnyWordSpecLike.scala:1219)
[info]   at org.scalatest.wordspec.AnyWordSpecLike.run$(AnyWordSpecLike.scala:44)
[info]   at akka.testkit.AkkaSpec.org$scalatest$BeforeAndAfterAll$$super$run(AkkaSpec.scala:54)
[info]   at org.scalatest.BeforeAndAfterAll.liftedTree1$1(BeforeAndAfterAll.scala:213)
[info]   at org.scalatest.BeforeAndAfterAll.run(BeforeAndAfterAll.scala:217)
[info]   at org.scalatest.BeforeAndAfterAll.run$(BeforeAndAfterAll.scala:135)
[info]   at akka.testkit.AkkaSpec.run(AkkaSpec.scala:54)
[info]   at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:318)
[info]   at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:513)
[info]   at sbt.ForkMain$Run.lambda$runTest$1(ForkMain.java:413)
[info]   at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[info]   at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[info]   at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[info]   at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by patriknw** [comment]
https://github.com/akka/akka/runs/5044975209?check_suite_focus=true#step:5:3464

**Comment by pvlugter** [comment]
https://github.com/akka/akka/runs/5225034447?check_suite_focus=true#step:5:3854

**Comment by patriknw** [comment]
https://github.com/akka/akka/runs/5245638646?check_suite_focus=true#step:5:3434

**Comment by octonato** [comment]
Closing for now. Probably fixed in #31147
